test/utils/asserts.hpp

```diff
@@ -42,14 +42,14 @@ MATCHER_P(Result, value, "")  // NOLINT(readability-redundant-string-init)
         if (arg.trapped || !arg.has_value)
             return false;
 
-        if constexpr (std::is_same_v<value_type, float>)
+        if constexpr (std::is_same_v<value_type, float> || std::is_same_v<value_type, test::FP32>)
         {
-            return arg.type == ValType::f32 && arg.value.f32 == test::FP{value};
+            return arg.type == ValType::f32 && test::FP32{arg.value.f32} == value;
```

Comment:
Not sure if similar change is needed for `CResult` matcher below.

Reply:
Changed in `CResult` too.